### PR TITLE
Issue #7 begging ui development

### PR DESCRIPTION
**Pull request by Asudano**

### CODE REVIEW

File: CharacterData.py [hunk 1]
```diff
@@ -1,225 +1,106 @@
...
+        self.__name = name
+        self.__game_class_options = []
+        self.__growth_rate_class = {}
+        self.__base_stats = []
+
+    @property
+    def name(self):
+        return self.__name
+
+    def add_base_stats(self, array):
```
Comment (kirons323) [inline CharacterData.py:54]: Was this supposed to have something in it?

File: CharacterData.py [hunk 2]
```diff
@@ -1,225 +1,106 @@
...
+
+        Args:
+            level : an int representing an in game level
+            game_class: a str describing an in game class
+
+        Returns:
+            A state object representing the average for this character at the
+                given class and level.
+        """
+        pass
```
Comment (kirons323) [inline CharacterData.py:92]: This is where the stuff from the Calculations.py should go

File: updateUi.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        self.charattribute.append(charname)
+
+    def createtextbox(self, attribute, gridN):
+        master = self.master
+        textbox = Entry(master)
+        textbox.grid(row=gridN, column=1, columnspan=1, sticky=E + W)
+        textbox.insert(0, attribute)
+        self.charattribute.append(textbox)
+
+    def createwidgets(self):
```
Comment (wagnet) [inline updateUi.py]: create_widgets

"""
    create_widgets creates display elements for update screen
"""

File: NewCharachterUi.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
+from tkinter import *
```
Comment (kirons323) [inline NewCharachterUi.py:1]: Will need documentation for each function

File: compareCharUi.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
+from tkinter import *
```
Comment (kirons323) [inline compareCharUi.py:1]: Will need documentation for each function

File: homescreen.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
+from tkinter import *
```
Comment (kirons323) [inline homescreen.py:1]: Will need documentation for each function

File: myteamUi.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
+from tkinter import *
```
Comment (kirons323) [inline myteamUi.py:2]: Will need documentation for each function

File: suggestteamUi.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
+from tkinter import *
```
Comment (kirons323) [inline suggestteamUi.py:1]: Will need documentation for each function

File: updateUi.py [hunk 9]
```diff
@@ -0,0 +1,92 @@
+from tkinter import *
```
Comment (kirons323) [inline updateUi.py:1]: Will need documentation for each function

File: visualizeprogressUI.py [hunk 10]
```diff
@@ -0,0 +1,45 @@
+from tkinter import *
```
Comment (kirons323) [inline visualizeprogressUI.py:1]: Will need documentation for each function

File: updateUi.py [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+        # 4. Add character Instance to user logs
+        self._user_logs.update_logs(char_instance)
+
+    def createlabels(self, charachter_name, grideN):
+        master = self.master
+        charname = Label(master, text=charachter_name)
+        charname.grid(row=grideN, column=0, columnspan=1, sticky=E + W)
+        self.charattribute.append(charname)
+
+    def createtextbox(self, attribute, gridN):
```
Comment (wagnet) [inline updateUi.py]: create_textbox

"""
create_textbox creates a text box element

Args:
attribute : str that specifies the attribute for text box
gridN : int that specifies number of rows
"""

File: updateUi.py [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+                     Stat.Def : int(self.charattribute[9].get()), Stat.Res : int(self.charattribute[10].get()), }
+        state = State(int(self.charattribute[2].get()), str(self.charattribute[1].get()), stat_dict)
+
+        # 3. Add state to character instance
+        char_instance.add_new_state(state)
+
+        # 4. Add character Instance to user logs
+        self._user_logs.update_logs(char_instance)
+
+    def createlabels(self, charachter_name, grideN):
```
Comment (wagnet) [inline updateUi.py]: create_labels(self, character_name, gridN)

"""
 create_labels creates a label for a UI element

Args:
character_name : str that specifies the name of the character
gridN : int that specifies the number of rows
"""

File: updateUi.py [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+
+
+class updateUi(Frame):
+    def __init__(self, user_logs, master=None):
+        Frame.__init__(self, master)
+        self.charattribute = []
+        self.createwidgets()
+        self.user_logs = user_logs
+
+    def update_f(self):
```
Comment (wagnet) [inline updateUi.py:17]: """
update_f updates a logged character with a new state given by the user
"""

File: updateUi.py [hunk 14]
```diff
@@ -0,0 +1,92 @@
+from tkinter import *
+from GameData import GameData
+from State import State
+from StatEnum import Stat
+from CharacterInstance import CharacterInstance
+from UserLogs import UserLogs
+
+
+class updateUi(Frame):
+    def __init__(self, user_logs, master=None):
```
Comment (wagnet) [inline updateUi.py:8]: """
inits a new UpdateUI object to handle updating of a CharacterInstance with a new state
"""

File: updateUi.py [hunk 15]
```diff
@@ -0,0 +1,92 @@
+from tkinter import *
+from GameData import GameData
+from State import State
+from StatEnum import Stat
+from CharacterInstance import CharacterInstance
+from UserLogs import UserLogs
+
+
+class updateUi(Frame):
```
Comment (wagnet) [inline updateUi.py]: UpdateUi

File: GameData.py [hunk 16]
```diff
@@ -1,232 +1,174 @@
...
+            infile = open(infile_name, "r")
+        except IOError:
+            # TODO: Make a better error handler
+            return
+
+        state = 0
+
+        for line in infile_name:
+            # separators
+            # Character,Class,HP GR,Str GR,Mag GR,Skl GR,Spd GR,Lck GR,Def GR,Res GR, - characters, their classes, and growth rates
```
Comment (wagnet) [inline GameData.py]: lines should have a max length of 80 characters

File: CharacterData.py [hunk 17]
```diff
@@ -1,225 +1,106 @@
...
+
+    def get_base_level(self):
```
Comment (wagnet) [inline CharacterData.py:94]: """
get_base_level fetches the level associated with base stats

Returns:
    int describing base level
"""

File: CharacterData.py [hunk 18]
```diff
@@ -1,225 +1,106 @@
...
+        return (self.base_stats)[0].level
```
Comment (wagnet) [inline CharacterData.py:95]: What is the [0] for?

File: CharacterData.py [hunk 19]
```diff
@@ -1,225 +1,106 @@
...
+
+    def get_base_stats(self):
```
Comment (wagnet) [inline CharacterData.py:97]: """
get_base_stats fetches the stats dictionary from the base stats object

Returns:
    a dict<Stat, float> describing the stats of a character when they are first encountered
"""

File: CharacterData.py [hunk 20]
```diff
@@ -1,225 +1,106 @@
...
+        return (self.base_stats)[0].get_stats()
+
+    def get_base_class(self):
```
Comment (wagnet) [inline CharacterData.py:100]: """
get_base_class fetches the game_class of a character when they are first encountered

Returns:
    a str describing the game_class of a character when they are first encountered
"""

File: CharacterData.py [hunk 21]
```diff
@@ -1,225 +1,106 @@
...
+        return (self.base_stats)[0].game_class
+
+    def add_class_and_growth_rates(self, game_class, rates):
```
Comment (wagnet) [inline CharacterData.py:103]: """
add_class_and_growth_rates adds the game class and growth rates for that character/class combination to the character's GameData object

Args:
game_class: a str describing an in game class that the character can be
rates: a dict<Stat, float> mapping in game stats to the values describing the growth rate of that stat when this character assumes that class
"""

File: StatEnum.py [hunk 22]
```diff
@@ -8,4 +8,17 @@ class Stat(Enum):
 	Spd = 4
 	Lck = 5
 	Def = 6
-	Res = 7
\ No newline at end of file
+	Res = 7
+
+def stat_dict(array):
+	# TODO: Add doc string
```
Comment (wagnet) [inline StatEnum.py:14]: """
   stat_dict makes a dictionary mapping stats onto values for those stats

Args:
    array : a list of floats giving values of the form
        [HP, Str, Mag, Skl, SPd, Lck, Def, Res]

Returns:
    a dict<Stat, float> describing values for game stats
"""

File: myteamUi.py [hunk 23]
```diff
@@ -0,0 +1,33 @@
...
+        Frame.__init__(self, master)
+        self.createwidgets()
+
+    def compare_char_f(self):
+        pass
+
+    def suggest_team_f(self):
+        pass
+
+    def createwidgets(self):
```
Comment (wagnet) [inline myteamUi.py:19]: create_widgets
"""
create_widgets creates display elements for myteam scren
 """

**Pull request merged**